src/main/java/ai/elimu/web/project/app_category/AppCategoryDeleteController.java

```diff
@@ -0,0 +1,36 @@
+package ai.elimu.web.project.app_category;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import ai.elimu.dao.AppCategoryDao;
+import ai.elimu.model.project.AppCategory;
+
+@Controller
+@RequestMapping("/project/{projectId}/app-category/delete")
+public class AppCategoryDeleteController {
+	
+private final Logger logger = Logger.getLogger(getClass());
+    
+    @Autowired
+    private AppCategoryDao appCategoryDao;
+    
+	@RequestMapping(value = "/{id}", method = RequestMethod.GET)
+    public String handleRequest(Model model, @PathVariable Long projectId, @PathVariable Long id) {
+        
+		logger.info("handleRequest");
+    	logger.info("projectId: " + projectId);
+        
+        AppCategory appCategory = appCategoryDao.read(id);
+		appCategoryDao.delete(appCategory);
```

Comment:
@raidenovski I tested the latest changes. When I'm pressing the "Delete" button, I get a DataIntegrityViolationException:

```
[ERROR] 2017-10-04 21:58:01,611 [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] - integrity constraint violation: foreign key no action; FK_ADQA65G6O4TI2CT3OQW0UW1BJ table: PROJECT_APPCATEGORY
[INFO] 2017-10-04 21:58:01,613 [org.hibernate.engine.jdbc.batch.internal.AbstractBatchImpl] - HHH000010: On release of batch it still contained JDBC statements
[ERROR] 2017-10-04 21:58:01,618 [ai.elimu.web.servlet.CustomSimpleMappingExceptionResolver] - Handler execution resulted in exception: org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [FK_ADQA65G6O4TI2CT3OQW0UW1BJ]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [FK_ADQA65G6O4TI2CT3OQW0UW1BJ]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:259)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:225)
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:521)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:761)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:730)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:484)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:291)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208)
	at com.sun.proxy.$Proxy69.delete(Unknown Source)
	at ai.elimu.web.project.app_category.AppCategoryDeleteController.handleRequest(AppCategoryDeleteController.java:30)
```

Are you getting the same error? And do you have any ideas how we could solve this?

Reply:
Yes, I will look into why this is happening